### PR TITLE
Migrate to --ssh-identity-file instead of --identity-file

### LINKED ISSUE #3434: Allow --ssh-identity-file or similar in knife bootstrap and knife ssh
User experience discrepancy: All of the `knife ssh` and `knife bootstrap` SSH long options begin with --ssh except one (--identity-file).

```
    -i IDENTITY_FILE,                The SSH identity file used for authentication
        --identity-file
...
    -G, --ssh-gateway GATEWAY        The ssh gateway
    -P, --ssh-password [PASSWORD]    The ssh password - will prompt if flag is specified but no password is given
    -p, --ssh-port PORT              The ssh port
    -x, --ssh-user USERNAME          The ssh username
...
```

### PR DESCRIPTION
**Pull request by jblaine**
Fixes #3434 

Introduces --ssh-identity-file which is more like the
other SSH arguments (they all begin with --ssh except
this one). Behavior is unchanged.

Migrates '-i' over with --ssh-identity-file

Retains --identity-file with a deprecation statement.

### CODE REVIEW

File: lib/chef/knife/bootstrap.rb [hunk 1]
```diff
@@ -366,7 +370,8 @@ def knife_ssh
         ssh.config[:ssh_port] = config[:ssh_port]
         ssh.config[:ssh_gateway] = config[:ssh_gateway]
         ssh.config[:forward_agent] = config[:forward_agent]
-        ssh.config[:identity_file] = config[:identity_file]
+        ssh.config[:identity_file] = config[:identity_file] # DEPRECATED
+        ssh.config[:ssh_identity_file] = config[:ssh_identity_file]
```
Comment (stevendanna) [inline lib/chef/knife/bootstrap.rb]: Perhaps we should *only* set `ssh.config[:ssh_identity_file] = config[:ssh_identity_file] || config[:identity_file]`

**Comment by stevendanna** [comment]
Are we sure this is all that is needed to get the chagne you want?  I would have assumed that you need to change the code where we create the ssh session here:

https://github.com/chef/chef/pull/3848/files#diff-db7525c27338848ec8b5972f6195ce77R222

**Comment by jblaine** [comment]
You're very right. My searching was hasty.

**Comment by jblaine** [comment]
```
% knife ssh dev 'echo $$' -m -i ~/.ssh/id_rsa
dev 4319
% knife ssh dev 'echo $$' -m --identity-file ~/.ssh/id_rsa
dev 4395
% knife ssh dev 'echo $$' -m --ssh-identity-file ~/.ssh/id_rsa
dev 4471
```

`knife bootstrap --ssh-user jblaine --sudo --ssh-identity-file ~/.ssh/id_rsa` worked as well.

**Comment by thommay** [comment]
Looks good, thanks! :+1:

**Comment by lamont-granquist** [comment]
:+1:

**Pull request closed without merging**